src/player.js

```diff
@@ -2123,6 +2123,7 @@ export default class Player extends FakeEventTarget {
     const track: ?Track = this._getTracksByType(type).find(track => Track.langComparer(language, track.language));
     if (track) {
       this.selectTrack(track);
+      this._markActiveTrack(track);
```

Comment:
consider adding protection for only calling `this._markActiveTrack(track);` if `track.active`, because if it is not active we will call the engine and after it changes the track we will get a track change event and call  `_markActiveTrack` from there.

Reply:
ok - so we now know we must call it as it registers some stuff in the engine - thanks @odedhutzler